src/diffusers/loaders.py

```diff
@@ -169,10 +170,11 @@ def load_attn_procs(self, pretrained_model_name_or_path_or_dict: Union[str, Dict
                         user_agent=user_agent,
                     )
                     state_dict = safetensors.torch.load_file(model_file, device="cpu")
-                except EnvironmentError:
+                except EnvironmentError as e:
+                    if use_safetensors is True:
+                        raise e
```

Comment:
It's potentially not the job of this PR because `EnvironmentError` was here before, but `EnvironmentError` is a very broad error e.g. anything io related. Can we add some docs on why it's ok to catch it here if we're using safe tensors?

Reply:
I can add some comment, but I don't know why it's `EnvironmentError`, I'm guessing it could be IOError since it's a file missing error, but maybe we also want to catch other types of "missing error".

Reply:
If it's an IOError because a file is missing, should we just explicitly check for the file not being present? cc @patrickvonplaten I think you committed this, do you have contet?